libazureinit/src/media.rs

```diff
@@ -187,6 +189,24 @@ pub fn parse_ovf_env(ovf_body: &str) -> Result<Environment, Error> {
     }
 }
 
+// Mount the given device, get OVF environment data, return it.
+pub fn mount_parse_ovf_env(dev: String) -> Result<Environment, anyhow::Error> {
+    let mount_media =
+        Media::new(PathBuf::from(dev), PathBuf::from(PATH_MOUNT_POINT));
+    let mounted = mount_media
+        .mount()
+        .with_context(|| "Failed to mount media.")?;
```

Comment:
Ideally the library should exclusively return errors of type `libazureinit::error::Error` and library users can use the types to handle different failure cases. If the client needs to react to mounting vs unmounting failures we could add some additional members to the enum, otherwise we can just propagate the error with `?`.

While the stack trace would make it clear which thing failed, this might also be a good opportunity to introduce a little logging. PR #83 is pulling in `tracing`, but it should be a minor conflict to `cargo add tracing` in this PR and add a `tracing::error!("Failed to mount media.")` call here.

If we're okay bumping the minimum Rust version to 1.76, you can do it like this:
```
let mounted = mount_media
    .mount()
    .inspect_err(|e| tracing::error!(error = ?e, "Failed to mount media."))?;
```

Otherwise, we can use `map_err` and just return the same error:
```
let mounted = mount_media
    .mount()
    .map_err(|e| {
        tracing::error!(error = ?e, "Failed to mount media.");
        e
    })?;
```

Reply:
Thanks, let's go for `inspect_err` with the minimum Rust 1.76.
It should be ok. Many distros seem to still have <= 1.75, but CI should run with the latest stable 1.78.